R/modifiers.R

```diff
@@ -202,6 +202,22 @@ type.stringr_fixed <- function(x, error_call = caller_env()) {
 }
 #' @export
 type.character <- function(x, error_call = caller_env()) {
+  if (length(x) == 1) {
+    if(is.na(x)) {
+      cli::cli_abort(
+        tr_("{.arg pattern} must be a string, not {.obj_type_friendly {x}}."),
+        call = error_call
+      )
+    }
+  }
```

Comment:
Do you think this branch is really needed? I think just the error below would be sufficient.